lib/array_intersection.rb

```diff
@@ -1,3 +1,11 @@
 def intersection(list1, list2)
-  
+  result = []
+  hash = {}
+  list1.each do |num|
+    hash[num] = true
+  end
+  list2.each do |num|
+    result << num if hash[num] == true
+  end
+  return result
```

Comment:
Adding some whitespace between the variable declarations, and also between the multi-line statements (the `each` loops) would help make this method easier to read.